### PR TITLE
Feature/marquee display thread

### PR DESCRIPTION
**Pull request by roshroff**
# Overview
This PR implements the skeleton for MarqueeDisplayThread. Right now this thread just prints a message on the console after every one sec. 
MarqueeDisplayThread uses the infrastructure provided by ITask.

# Testing
The Blinker and DataAcquistion tasks still run and perform as expected.

Console log output:
Serial console initialized
Now starting task 'BlinkerTask'
Now starting task 'DataAcqusitionTask'
Now starting task 'MarqueeDisplayThread'
FreeRTOS Scheduler Running...
ENTER: virtual void BlinkerTask::Run()
ENTER: virtual void DataAcquisitionTask::Run()
[000014912] Hello from virtual void DataAcquisitionTask::Run()
ENTER: virtual void MarqueeDisplayThread::Run()
[000024516] Hello from virtual void MarqueeDisplayThread::Run()
[001062428] Hello from virtual void DataAcquisitionTask::Run()
[001079436] Hello from virtual void MarqueeDisplayThread::Run()
[002117084] Hello from virtual void DataAcquisitionTask::Run()
[002134096] Hello from virtual void MarqueeDisplayThread::Run()
[003171728] Hello from virtual void DataAcquisitionTask::Run()
[003188744] Hello from virtual void MarqueeDisplayThread::Run()
Button Pressed! virtual void BlinkerTask::Run()
Button Pressed! virtual void BlinkerTask::Run()
[004226388] Hello from virtual void DataAcquisitionTask::Run()
[004243400] Hello from virtual void MarqueeDisplayThread::Run()
[005281056] Hello from virtual void DataAcquisitionTask::Run()
[005298068] Hello from virtual void MarqueeDisplayThread::Run()
Button Pressed! virtual void BlinkerTask::Run()
[006335748] Hello from virtual void DataAcquisitionTask::Run()
[006352760] Hello from virtual void MarqueeDisplayThread::Run()
[007390424] Hello from virtual void DataAcquisitionTask::Run()
[007407444] Hello from virtual void MarqueeDisplayThread::Run()
Button Pressed! virtual void BlinkerTask::Run()

### CODE REVIEW

File: arduino/KnexElevator/KnexElevator.ino [hunk 1]
```diff
@@ -60,10 +62,12 @@ void setup(void)
     // Initialize application tasks
     gvBlinkerTask.Initialize();
     gvDataAcquisitionTask.Initialize();
+    gvMarqueeDisplayThread.Initialize();
```
Comment (saunmeht) [inline arduino/KnexElevator/KnexElevator.ino]: Let's call it Task instead of Thread.

File: arduino/KnexElevator/MarqueeDisplayThread.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+// Include(s)
+// /////////////////////////////////////////////////////////////////////////////
+#include <Arduino_FreeRTOS.h>
+
+#include "ITask.h"
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief
+/// A thread for displaying information on the console.
```
Comment (gfarrand) [inline arduino/KnexElevator/MarqueeDisplayThread.h]: Please update this description showing what this thread will be doing...
It updates the marquee display, showing which floor each elevator is on.

File: arduino/KnexElevator/MarqueeDisplayThread.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
+// /////////////////////////////////////////////////////////////////////////////
+//
+// MIT License
+//
+// Copyright (c) 2017 gfarrand (grant.farrand@gmail.com)
```
Comment (DizzyBytes) [inline arduino/KnexElevator/MarqueeDisplayThread.cpp]: Leave the author name as is in the copyright?

**Comment by roshroff** [comment]
Addressed review comments. Changes pushed out.

**Review by gfarrand — APPROVED** [review]
Looks good!

**Pull request merged**